openmicroscopy/openmicroscopy
# Add CLI info, listusers and listgroups subcommands

Fixes https://trac.openmicroscopy.org/ome/ticket/12589

This PR adds 
- a `info` subcommand to both `UserControl` and `GroupControl` plugins. These commands have the same behavior as the `list` subcommand for a subset of users/groups. The following commands should now be accepted

  ```
bin/omero user info   # Give info about the context user
bin/omero user info user-1
bin/omero user info user-1 --user-id 3 --user-name user-4
bin/omero group info   # Give info about the context user
bin/omero group info read-write1
bin/omero group info read-write1 --group-id 3 --group-name read-annotate-1
```

- a `bin/omero group listusers` and a `bin/omero user listgroups` subcommands. Both subcommands either take no argument (using the context) or a single argument. The following calls should work:

  ```
bin/omero group listusers   # Return the users of the context group
bin/omero group listusers read-write1
bin/omero user listgroups   # Return the groups of the context user
bin/omero user listgroups user-6
```

To test this PR:
- check the commands above are working in a multi-user/multi-group context
- check the integration tests are passing both for the new commands as well as the `user list` and `group list` subcommands which were refactored as part of this PR

--no-rebase
